### ISSUE TITLE
Strange cartridge issue [V1.38]

### ISSUE THREAD

**Issue by RichardTND** [GideonZ/1541ultimate#176]
ULTIMATE 64: Firmware V1.38 ... Some games with cartridge protection detect cartridges and fails on running games properly - even when no internal/external cartridge is enabled. Example: My legal tape original of Great Gurianos by Elite Software runs the program after loading the last block. However, the game cannot be played. A message "REMOVE CARTRIDGE AND RELOAD" scrolls across locking the game from being played.

**Comment by GideonZ** [comment]
That is quite interesting indeed. I'd love to see the code that performs
this test, so I can figure out what it sees.

**Comment by xlar54** [comment]
does it do this in VICE as well?  if so, the TAP file would be useful in diagnosing the issue

**Comment by RichardTND** [comment]
It works in VICE with Action Replay plugged in and unplugged if "fast load option (F7)" is not used. Otherwise like with the Ultimate 64, the game is locked.

**Comment by p2mate** [comment]
similar to #205 ?

**Comment by GideonZ** [comment]
Yeah, seems to be all the same problem indeed. :-)  The solution is already known, but just needs to be implemented and tested.
BTW, @p2mate , since you have provided patches to some titles, maybe you could help to extract some failing code from another demo, in such a way that a PRG file is created that exposes the problem right from the start?

**Comment by p2mate** [comment]
I will try to do that.

**Comment by p2mate** [comment]
[snap_cart_detect.zip](https://github.com/GideonZ/1541ultimate/files/6135646/snap_cart_detect.zip)

Barry McGuigan World Championship Boxing_MRZ detection code reimplemented as a standalone prg. Only tested on vice and 1.37 U64 so far. I don't actually own a real C64 anymore.

**Comment by p2mate** [comment]
@GideonZ which demo is this?

**Comment by GideonZ** [comment]
@p2mate The demo is vampy. I am on it already. And it is not related to this issue.

**Comment by RichardTND** [comment]
The "Remove Cartridge and Reload" issue after loading Great Gurianos has been removed in firmware V3.15A. Great Gurianos works without that error.
